lib/CAPI/Dialect/OM.cpp

```diff
@@ -221,6 +230,50 @@ OMEvaluatorValue omEvaluatorTupleGetElement(OMEvaluatorValue evaluatorValue,
                   ->getElements()[pos]);
 }
 
+/// Get an element of the Map.
+OMEvaluatorValue omEvaluatorMapGetElement(OMEvaluatorValue evaluatorValue,
+                                          MlirAttribute attr) {
+  const auto &elements =
+      cast<evaluator::MapValue>(unwrap(evaluatorValue).get())->getElements();
+  const auto &it = elements.find(unwrap(attr));
+  if (it != elements.end())
+    return wrap(it->second);
+  return OMEvaluatorValue{nullptr};
+}
+
+OMEvaluatorValue omEvaluatorMapGetElementByInt(OMEvaluatorValue evaluatorValue,
+                                               intptr_t attr) {
+  auto *ptr = cast<evaluator::MapValue>(unwrap(evaluatorValue).get());
+  auto keyType = ptr->getType().cast<MapType>().getKeyType();
+
+  // Reject if the key type is not integer.
+  if (!isa<IntegerType>(keyType))
+    return OMEvaluatorValue{nullptr};
+
+  auto keyAttr = mlir::IntegerAttr::get(keyType, attr);
+  const auto &it = ptr->getElements().find(keyAttr);
+  if (it != ptr->getElements().end())
+    return wrap(it->second);
+
+  return OMEvaluatorValue{nullptr};
+}
+
+OMEvaluatorValue omEvaluatorMapGetElementByStr(OMEvaluatorValue evaluatorValue,
+                                               MlirStringRef attr) {
+  auto *ptr = cast<evaluator::MapValue>(unwrap(evaluatorValue).get());
+  auto strType = StringType::get(ptr->getType().getContext());
+  const auto &it =
+      ptr->getElements().find(StringAttr::get(unwrap(attr), strType));
+  if (it != ptr->getElements().end())
+    return wrap(it->second);
+  return OMEvaluatorValue{nullptr};
+}
```

Comment:
Do we really need the int and str specializations ? Isn't `omEvaluatorMapGetElement` enough ?

Reply:
I had this question too. My suggestion is to have one omEvaluatorMapGetElement in the C API, which takes an MlirAttribute, and then in `Map::lookup` in the pybind module, instead of calling one of the current three C APIs, if an int is passed, use the C API to construct an int attribute, and if a string is passed, use the C API to construct a string attribute. Then call the one API that just takes an attribute.